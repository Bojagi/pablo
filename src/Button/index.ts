import './styles';

export * from './Button';
