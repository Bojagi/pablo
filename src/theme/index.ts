export * from './defaultTheme';
export * from './context';
export * from './PabloThemeProvider';
export * from './useComponentStyle';
