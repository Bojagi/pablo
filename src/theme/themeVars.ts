import isObject from 'isobject';
import { breakpointVars } from './breakpoints';
import { colorVars } from './colors';
import { typographyVars } from './typography';

export const themeVarNames = {
  typography: typographyVars,
  colors: colorVars,
  breakpoints: breakpointVars,
};

function createThemeReferences<T extends Record<string, any>>(obj: T): T {
  return Object.entries(obj).reduce(
    (acc, [key, value]) => ({
      ...acc,
      [key]: isObject(value) ? createThemeReferences(value as T) : `var(--${value})`,
    }),
    {} as any
  );
}

export const themeVars = createThemeReferences(themeVarNames);
