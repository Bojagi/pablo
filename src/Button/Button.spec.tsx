import { cleanup, render } from '@testing-library/react';
import React from 'react';
import { PabloThemeProvider } from '../theme';
import { themeVars } from '../theme/themeVars';
import { Button } from './Button';

afterEach(cleanup);

test.each([
  [
    'primary',
    'brand',
    {
      color: themeVars.colors.brand.contrastText,
      backgroundColor: themeVars.colors.brand.main,
      borderColor: themeVars.colors.brand.main,
      hoverBackgroundColor: themeVars.colors.brand.dark,
      hoverBorderColor: themeVars.colors.brand.dark,
    },
  ],
  [
    'primaryInverted',
    'brand',
    {
      color: themeVars.colors.brand.main,
      backgroundColor: themeVars.colors.brand.contrastText,
      borderColor: themeVars.colors.brand.contrastText,
      hoverBackgroundColor: themeVars.colors.brand.lightest,
      hoverBorderColor: themeVars.colors.brand.lightest,
    },
  ],
  [
    'secondary',
    'brand',
    {
      color: themeVars.colors.brand.dark,
      backgroundColor: 'transparent',
      borderColor: themeVars.colors.brand.dark,
      hoverBackgroundColor: themeVars.colors.brand.lightest,
      hoverBorderColor: themeVars.colors.brand.darkest,
    },
  ],
  [
    'secondaryInverted',
    'brand',
    {
      color: themeVars.colors.brand.contrastText,
      backgroundColor: 'transparent',
      borderColor: themeVars.colors.brand.contrastText,
      hoverBackgroundColor: themeVars.colors.brand.dark,
      hoverBorderColor: themeVars.colors.brand.contrastText,
    },
  ],
  [
    'text',
    'brand',
    {
      color: themeVars.colors.brand.dark,
      backgroundColor: 'transparent',
      borderColor: 'transparent',
      hoverBackgroundColor: themeVars.colors.brand.lightest,
      hoverBorderColor: 'transparent',
    },
  ],
  [
    'textInverted',
    'brand',
    {
      color: themeVars.colors.brand.contrastText,
      backgroundColor: 'transparent',
      borderColor: 'transparent',
      hoverBackgroundColor: themeVars.colors.brand.dark,
      hoverBorderColor: 'transparent',
    },
  ],
  [
    'primary',
    'plain',
    {
      color: themeVars.colors.common.black,
      backgroundColor: themeVars.colors.gray['50'],
      borderColor: themeVars.colors.gray['100'],
      hoverBackgroundColor: themeVars.colors.gray['100'],
      hoverBorderColor: themeVars.colors.gray['200'],
    },
  ],
  [
    'primaryInverted',
    'plain',
    {
      color: themeVars.colors.common.black,
      backgroundColor: themeVars.colors.common.white,
      borderColor: themeVars.colors.common.white,
      hoverBackgroundColor: themeVars.colors.gray['50'],
      hoverBorderColor: themeVars.colors.gray['50'],
    },
  ],
  [
    'secondary',
    'plain',
    {
      color: themeVars.colors.common.black,
      backgroundColor: 'transparent',
      borderColor: themeVars.colors.common.black,
      hoverBackgroundColor: themeVars.colors.gray['50'],
      hoverBorderColor: themeVars.colors.common.black,
    },
  ],
  [
    'secondaryInverted',
    'plain',
    {
      color: themeVars.colors.common.white,
      backgroundColor: 'transparent',
      borderColor: themeVars.colors.common.white,
      hoverBackgroundColor: themeVars.colors.gray['800'],
      hoverBorderColor: themeVars.colors.common.white,
    },
  ],
  [
    'text',
    'plain',
    {
      color: themeVars.colors.common.black,
      backgroundColor: 'transparent',
      borderColor: 'transparent',
      hoverBackgroundColor: themeVars.colors.gray['50'],
      hoverBorderColor: 'transparent',
    },
  ],
  [
    'textInverted',
    'plain',
    {
      color: themeVars.colors.common.white,
      backgroundColor: 'transparent',
      borderColor: 'transparent',
      hoverBackgroundColor: themeVars.colors.gray['800'],
      hoverBorderColor: 'transparent',
    },
  ],
  [
    'primary',
    'negative',
    {
      color: themeVars.colors.negative.contrastText,
      backgroundColor: themeVars.colors.negative.main,
      borderColor: themeVars.colors.negative.main,
      hoverBackgroundColor: themeVars.colors.negative.dark,
      hoverBorderColor: themeVars.colors.negative.dark,
    },
  ],
  [
    'primaryInverted',
    'negative',
    {
      color: themeVars.colors.negative.main,
      backgroundColor: themeVars.colors.negative.contrastText,
      borderColor: themeVars.colors.negative.contrastText,
      hoverBackgroundColor: themeVars.colors.negative.lightest,
      hoverBorderColor: themeVars.colors.negative.lightest,
    },
  ],
  [
    'secondary',
    'negative',
    {
      color: themeVars.colors.negative.dark,
      backgroundColor: 'transparent',
      borderColor: themeVars.colors.negative.dark,
      hoverBackgroundColor: themeVars.colors.negative.lightest,
      hoverBorderColor: themeVars.colors.negative.darkest,
    },
  ],
  [
    'secondaryInverted',
    'negative',
    {
      color: themeVars.colors.negative.contrastText,
      backgroundColor: 'transparent',
      borderColor: themeVars.colors.negative.contrastText,
      hoverBackgroundColor: themeVars.colors.negative.dark,
      hoverBorderColor: themeVars.colors.negative.contrastText,
    },
  ],
  [
    'text',
    'negative',
    {
      color: themeVars.colors.negative.dark,
      backgroundColor: 'transparent',
      borderColor: 'transparent',
      hoverBackgroundColor: themeVars.colors.negative.lightest,
      hoverBorderColor: 'transparent',
    },
  ],
  [
    'textInverted',
    'negative',
    {
      color: themeVars.colors.negative.contrastText,
      backgroundColor: 'transparent',
      borderColor: 'transparent',
      hoverBackgroundColor: themeVars.colors.negative.dark,
      hoverBorderColor: 'transparent',
    },
  ],
  [
    'primary',
    'positive',
    {
      color: themeVars.colors.positive.contrastText,
      backgroundColor: themeVars.colors.positive.main,
      borderColor: themeVars.colors.positive.main,
      hoverBackgroundColor: themeVars.colors.positive.dark,
      hoverBorderColor: themeVars.colors.positive.dark,
    },
  ],
  [
    'primaryInverted',
    'positive',
    {
      color: themeVars.colors.positive.main,
      backgroundColor: themeVars.colors.positive.contrastText,
      borderColor: themeVars.colors.positive.contrastText,
      hoverBackgroundColor: themeVars.colors.positive.lightest,
      hoverBorderColor: themeVars.colors.positive.lightest,
    },
  ],
  [
    'secondary',
    'positive',
    {
      color: themeVars.colors.positive.dark,
      backgroundColor: 'transparent',
      borderColor: themeVars.colors.positive.dark,
      hoverBackgroundColor: themeVars.colors.positive.lightest,
      hoverBorderColor: themeVars.colors.positive.darkest,
    },
  ],
  [
    'secondaryInverted',
    'positive',
    {
      color: themeVars.colors.positive.contrastText,
      backgroundColor: 'transparent',
      borderColor: themeVars.colors.positive.contrastText,
      hoverBackgroundColor: themeVars.colors.positive.dark,
      hoverBorderColor: themeVars.colors.positive.contrastText,
    },
  ],
  [
    'text',
    'positive',
    {
      color: themeVars.colors.positive.dark,
      backgroundColor: 'transparent',
      borderColor: 'transparent',
      hoverBackgroundColor: themeVars.colors.positive.lightest,
      hoverBorderColor: 'transparent',
    },
  ],
  [
    'textInverted',
    'positive',
    {
      color: themeVars.colors.positive.contrastText,
      backgroundColor: 'transparent',
      borderColor: 'transparent',
      hoverBackgroundColor: themeVars.colors.positive.dark,
      hoverBorderColor: 'transparent',
    },
  ],
])('render %s %s color button', (variant, color, expectedStyles) => {
  const { container, getByTestId } = renderComponent({ variant, color });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-button')).toBeEnabled();
  expect(getByTestId('pbl-button')).toHaveStyleRule('background', expectedStyles.backgroundColor);
  expect(getByTestId('pbl-button')).toHaveStyleRule('color', expectedStyles.color);
  expect(getByTestId('pbl-button')).toHaveStyleRule('border-color', expectedStyles.borderColor);
  expect(getByTestId('pbl-button')).toHaveStyleRule(
    'border-color',
    expectedStyles.hoverBorderColor,
    {
      modifier: ':hover:enabled',
    }
  );
  expect(getByTestId('pbl-button')).toHaveStyleRule(
    'background',
    expectedStyles.hoverBackgroundColor,
    {
      modifier: ':hover:enabled',
    }
  );
});

test('Render primary brand button when no variant or color is given', () => {
  const expectedStyles = {
    color: themeVars.colors.brand.contrastText,
    backgroundColor: themeVars.colors.brand.main,
    borderColor: themeVars.colors.brand.main,
    hoverBackgroundColor: themeVars.colors.brand.dark,
    hoverBorderColor: themeVars.colors.brand.dark,
  };

  const { getByTestId } = renderComponent({});
  expect(getByTestId('pbl-button')).toBeEnabled();
  expect(getByTestId('pbl-button')).toHaveStyleRule('background', expectedStyles.backgroundColor);
  expect(getByTestId('pbl-button')).toHaveStyleRule('color', expectedStyles.color);
  expect(getByTestId('pbl-button')).toHaveStyleRule('border-color', expectedStyles.borderColor);
  expect(getByTestId('pbl-button')).toHaveStyleRule(
    'border-color',
    expectedStyles.hoverBorderColor,
    {
      modifier: ':hover:enabled',
    }
  );
  expect(getByTestId('pbl-button')).toHaveStyleRule(
    'background',
    expectedStyles.hoverBackgroundColor,
    {
      modifier: ':hover:enabled',
    }
  );
});

test('Render primary brand button when unknown variant is given', () => {
  const expectedStyles = {
    color: themeVars.colors.brand.contrastText,
    backgroundColor: themeVars.colors.brand.main,
    borderColor: themeVars.colors.brand.main,
    hoverBackgroundColor: themeVars.colors.brand.dark,
    hoverBorderColor: themeVars.colors.brand.dark,
  };

  const { getByTestId } = renderComponent({ variant: 'unknown' });
  expect(getByTestId('pbl-button')).toBeEnabled();
  expect(getByTestId('pbl-button')).toHaveStyleRule('background', expectedStyles.backgroundColor);
  expect(getByTestId('pbl-button')).toHaveStyleRule('color', expectedStyles.color);
  expect(getByTestId('pbl-button')).toHaveStyleRule('border-color', expectedStyles.borderColor);
  expect(getByTestId('pbl-button')).toHaveStyleRule(
    'border-color',
    expectedStyles.hoverBorderColor,
    {
      modifier: ':hover:enabled',
    }
  );
  expect(getByTestId('pbl-button')).toHaveStyleRule(
    'background',
    expectedStyles.hoverBackgroundColor,
    {
      modifier: ':hover:enabled',
    }
  );
});

test('render button with full width', () => {
  const { getByTestId } = renderComponent({ fullWidth: true });
  expect(getByTestId('pbl-button')).toHaveStyleRule('width', '100%');
});

test('render button with start icon', () => {
  const { container, getByTestId } = renderComponent({ startIcon: <div>myIcon</div> });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-button-icon')).toBeDefined();
  expect(getByTestId('pbl-button-icon')).toHaveStyleRule('margin-right', '8px');
});

test('render button with end icon', () => {
  const { container, getByTestId } = renderComponent({ endIcon: <div>myIcon</div> });
  expect(container).toMatchSnapshot();
  expect(getByTestId('pbl-button-icon')).toBeDefined();
  expect(getByTestId('pbl-button-icon')).toHaveStyleRule('margin-left', '8px');
});

test('disable button element when disabled prop is true', () => {
  const { getByTestId } = renderComponent({ disabled: true });
  expect(getByTestId('pbl-button')).toBeDisabled();
  expect(getByTestId('pbl-button')).toHaveStyleRule('opacity', '0.4', {
    modifier: ':disabled',
  });
});

function renderComponent(props) {
  return render(
    <PabloThemeProvider>
      <Button {...props}>Hello</Button>
    </PabloThemeProvider>
  );
}
