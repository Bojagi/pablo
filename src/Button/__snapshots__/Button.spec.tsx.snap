// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render button with icon 1`] = `
.c2 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  margin-right: 8px;
}

.c0 {
  color: light:#EBE7FF;
  main: #6A50F2;
  dark: #301A9E;
  contrast-text: #ffffff;
  contrast-text-light: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  -webkit-transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  outline: none;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c3 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.29em;
  margin-bottom: 0;
}

.c1 {
  color: #ffffff;
  background: #6A50F2;
  border-color: #6A50F2;
}

.c1:focus {
  box-shadow: 0 0 0 4px #EBE7FF;
}

.c1:hover:not(:disabled) {
  background: #301A9E;
  border-color: #301A9E;
}

<div>
  <button
    class="c0 c1"
    color="brand"
    data-testid="pbl-button"
  >
    <div
      class="c2"
      data-testid="pbl-button-icon"
      display="flex"
    >
      <div>
        myIcon
      </div>
    </div>
    <span
      class="c3"
    >
      Hello
    </span>
  </button>
</div>
`;

exports[`render primary brand color button 1`] = `
.c0 {
  color: light:#EBE7FF;
  main: #6A50F2;
  dark: #301A9E;
  contrast-text: #ffffff;
  contrast-text-light: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  -webkit-transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  outline: none;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.29em;
  margin-bottom: 0;
}

.c1 {
  color: #ffffff;
  background: #6A50F2;
  border-color: #6A50F2;
}

.c1:focus {
  box-shadow: 0 0 0 4px #EBE7FF;
}

.c1:hover:not(:disabled) {
  background: #301A9E;
  border-color: #301A9E;
}

<div>
  <button
    class="c0 c1"
    color="brand"
    data-testid="pbl-button"
  >
    <span
      class="c2"
    >
      Hello
    </span>
  </button>
</div>
`;

exports[`render primary negative color button 1`] = `
.c0 {
  color: light:#FFD3D0;
  main: #F44336;
  dark: #C92C21;
  contrast-text: #ffffff;
  contrast-text-light: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  -webkit-transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  outline: none;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.29em;
  margin-bottom: 0;
}

.c1 {
  color: #ffffff;
  background: #F44336;
  border-color: #F44336;
}

.c1:focus {
  box-shadow: 0 0 0 4px #FFD3D0;
}

.c1:hover:not(:disabled) {
  background: #C92C21;
  border-color: #C92C21;
}

<div>
  <button
    class="c0 c1"
    color="negative"
    data-testid="pbl-button"
  >
    <span
      class="c2"
    >
      Hello
    </span>
  </button>
</div>
`;

exports[`render primary plain color button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  -webkit-transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  outline: none;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.29em;
  margin-bottom: 0;
}

.c1 {
  color: #000000;
  background: #f0f0f0;
  border-color: #e0e0e0;
}

.c1:focus {
  box-shadow: 0 0 0 4px #f0f0f0;
}

.c1:hover:not(:disabled) {
  background: #e0e0e0;
  border-color: #d0d0d0;
}

<div>
  <button
    class="c0 c1"
    color="plain"
    data-testid="pbl-button"
  >
    <span
      class="c2"
    >
      Hello
    </span>
  </button>
</div>
`;

exports[`render primary positive color button 1`] = `
.c0 {
  color: light:#C7F8D6;
  main: #35D968;
  dark: #30C35E;
  contrast-text: #ffffff;
  contrast-text-light: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  -webkit-transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  outline: none;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.29em;
  margin-bottom: 0;
}

.c1 {
  color: #ffffff;
  background: #35D968;
  border-color: #35D968;
}

.c1:focus {
  box-shadow: 0 0 0 4px #C7F8D6;
}

.c1:hover:not(:disabled) {
  background: #30C35E;
  border-color: #30C35E;
}

<div>
  <button
    class="c0 c1"
    color="positive"
    data-testid="pbl-button"
  >
    <span
      class="c2"
    >
      Hello
    </span>
  </button>
</div>
`;

exports[`render secondary brand color button 1`] = `
.c0 {
  color: light:#EBE7FF;
  main: #6A50F2;
  dark: #301A9E;
  contrast-text: #ffffff;
  contrast-text-light: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  -webkit-transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  outline: none;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.29em;
  margin-bottom: 0;
}

.c1 {
  color: #6A50F2;
  border-color: #6A50F2;
}

.c1:focus {
  box-shadow: 0 0 0 4px #EBE7FF;
}

.c1:hover:not(:disabled) {
  color: #301A9E;
  background: #EBE7FF;
}

<div>
  <button
    class="c0 c1"
    color="brand"
    data-testid="pbl-button"
  >
    <span
      class="c2"
    >
      Hello
    </span>
  </button>
</div>
`;

exports[`render secondary negative color button 1`] = `
.c0 {
  color: light:#FFD3D0;
  main: #F44336;
  dark: #C92C21;
  contrast-text: #ffffff;
  contrast-text-light: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  -webkit-transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  outline: none;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.29em;
  margin-bottom: 0;
}

.c1 {
  color: #F44336;
  border-color: #F44336;
}

.c1:focus {
  box-shadow: 0 0 0 4px #FFD3D0;
}

.c1:hover:not(:disabled) {
  color: #C92C21;
  background: #FFD3D0;
}

<div>
  <button
    class="c0 c1"
    color="negative"
    data-testid="pbl-button"
  >
    <span
      class="c2"
    >
      Hello
    </span>
  </button>
</div>
`;

exports[`render secondary plain color button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  -webkit-transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  outline: none;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.29em;
  margin-bottom: 0;
}

.c1 {
  color: #000000;
  border-color: #000000;
}

.c1:focus {
  box-shadow: 0 0 0 4px #f0f0f0;
}

.c1:hover:not(:disabled) {
  background: #f0f0f0;
}

<div>
  <button
    class="c0 c1"
    color="plain"
    data-testid="pbl-button"
  >
    <span
      class="c2"
    >
      Hello
    </span>
  </button>
</div>
`;

exports[`render secondary positive color button 1`] = `
.c0 {
  color: light:#C7F8D6;
  main: #35D968;
  dark: #30C35E;
  contrast-text: #ffffff;
  contrast-text-light: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  -webkit-transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  outline: none;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.29em;
  margin-bottom: 0;
}

.c1 {
  color: #35D968;
  border-color: #35D968;
}

.c1:focus {
  box-shadow: 0 0 0 4px #C7F8D6;
}

.c1:hover:not(:disabled) {
  color: #30C35E;
  background: #C7F8D6;
}

<div>
  <button
    class="c0 c1"
    color="positive"
    data-testid="pbl-button"
  >
    <span
      class="c2"
    >
      Hello
    </span>
  </button>
</div>
`;

exports[`render text brand color button 1`] = `
.c0 {
  color: light:#EBE7FF;
  main: #6A50F2;
  dark: #301A9E;
  contrast-text: #ffffff;
  contrast-text-light: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  -webkit-transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  outline: none;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.29em;
  margin-bottom: 0;
}

.c1 {
  color: #6A50F2;
}

.c1:focus {
  box-shadow: 0 0 0 4px #EBE7FF;
}

.c1:hover:not(:disabled) {
  color: #301A9E;
  background: #EBE7FF;
}

<div>
  <button
    class="c0 c1"
    color="brand"
    data-testid="pbl-button"
  >
    <span
      class="c2"
    >
      Hello
    </span>
  </button>
</div>
`;

exports[`render text negative color button 1`] = `
.c0 {
  color: light:#FFD3D0;
  main: #F44336;
  dark: #C92C21;
  contrast-text: #ffffff;
  contrast-text-light: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  -webkit-transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  outline: none;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.29em;
  margin-bottom: 0;
}

.c1 {
  color: #F44336;
}

.c1:focus {
  box-shadow: 0 0 0 4px #FFD3D0;
}

.c1:hover:not(:disabled) {
  color: #C92C21;
  background: #FFD3D0;
}

<div>
  <button
    class="c0 c1"
    color="negative"
    data-testid="pbl-button"
  >
    <span
      class="c2"
    >
      Hello
    </span>
  </button>
</div>
`;

exports[`render text plain color button 1`] = `
.c0 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  -webkit-transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  outline: none;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.29em;
  margin-bottom: 0;
}

.c1 {
  color: #000000;
}

.c1:focus {
  box-shadow: 0 0 0 4px #f0f0f0;
}

.c1:hover:not(:disabled) {
  background: #f0f0f0;
}

<div>
  <button
    class="c0 c1"
    color="plain"
    data-testid="pbl-button"
  >
    <span
      class="c2"
    >
      Hello
    </span>
  </button>
</div>
`;

exports[`render text positive color button 1`] = `
.c0 {
  color: light:#C7F8D6;
  main: #35D968;
  dark: #30C35E;
  contrast-text: #ffffff;
  contrast-text-light: #000000;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-direction: row;
  -ms-flex-direction: row;
  flex-direction: row;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 4px 8px;
  border: 1px solid transparent;
  background: transparent;
  border-radius: 4px;
  -webkit-transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  transition: background-color,0.3s,border-color 0.3s,box-shadow 0.3s ease-in-out;
  outline: none;
}

.c0:not(:disabled) {
  cursor: pointer;
}

.c0:disabled {
  opacity: 0.4;
  cursor: normal;
}

.c2 {
  font-family: "IBM Plex Sans",sans-serif;
  font-style: normal;
  font-weight: normal;
  margin: 0;
  font-size: 0.875rem;
  line-height: 1.29em;
  margin-bottom: 0;
}

.c1 {
  color: #35D968;
}

.c1:focus {
  box-shadow: 0 0 0 4px #C7F8D6;
}

.c1:hover:not(:disabled) {
  color: #30C35E;
  background: #C7F8D6;
}

<div>
  <button
    class="c0 c1"
    color="positive"
    data-testid="pbl-button"
  >
    <span
      class="c2"
    >
      Hello
    </span>
  </button>
</div>
`;
